### PR TITLE
Add developer console page

### PR DESCRIPTION
**Pull request by csauve**
The main purpose of this PR is to add a page for developer console commands. What classifies as a developer console command? Well, anything that's not found in the BSL bible and hs_doc.txt output. This closes the gap significantly, but we are still missing some Sapien-specific commands like radiosity and dedicated server commands. I'd like to fill those in over time, as well as docs for all of these developer commands, so this is just a starting point.

There are nearly 500 new functions added in this PR. Unlike hs_doc, these are pretty much undocumented. I've tested and documented 88 of them. While testing, I came across some things that needed inclusion or correcting in our other pages, so you will see some other changes than just the new developer-console page.

### CODE REVIEW

File: src/content/h1/engine/developer-console/debug.yml [hunk 1]
```diff
@@ -0,0 +1,3195 @@
...
+          for the first person arms and weapon. Defaults to `1024.0`.
+          The world clipping distance can be set with `rasterizer_far_clip_distance`.
+    - slug: rasterizer_first_person_weapon_near_clip_distance
+      info:
+        en: |-
+          ```hsc
+          (rasterizer_first_person_weapon_near_clip_distance [real])
+          ```
+          Sets the [near clip distance](https://en.wikipedia.org/wiki/Clipping_(computer_graphics))
+          of the first person arms and weapon. Defaults to `0.0.011719`.
```
Comment (Mimickal) [inline src/content/h1/engine/developer-console/debug.yml:2615]: `0.0.011719`? Is that right or is there an extra `0.` in there?

File: src/content/h1/engine/renderer/readme.md [hunk 2]
```diff
@@ -7,7 +7,14 @@
   </figcaption>
 </figure>
 
-The **renderer** or **rasterizer** is the system of [Halo's engine][engine] responsible for drawing the scene to the screen. Each [ported edition][h1] of Halo has a slightly different renderer in terms of how well it reproduces the classic Xbox appearance. Halo uses the DirectX 9 API and shader version 2.0, being an early adopter of programmable shaders. Since support in user hardware was not as widespread as today, the renderer can be configured with [argumements][arguments#graphics-options] to use older shader versions or even [fixed function][ff] compatibility.
+The **renderer** or **rasterizer** is the system of [Halo's engine][engine] responsible for drawing the scene to the screen. Each [ported edition][h1] of Halo has a slightly different renderer in terms of how well it reproduces the classic Xbox appearance.
+
+Gearbox-era Halo uses the DirectX 9 API and shader version 2.0, being an early adopter of programmable shaders. Since support in user hardware was not as widespread as today, the renderer can be configured with [argumements][arguments#graphics-options] to use older shader versions or even [fixed function][ff] compatibility.
```
Comment (Mimickal) [inline src/content/h1/engine/renderer/readme.md:12]: "argumements"

File: src/content/h1/engine/scripting/hsc.yml [hunk 3]
```diff
@@ -1748,15 +1749,15 @@ Functions:
         en: |-
           ```hsc
           (camera_set_dead <unit>)
-          (camera_set_dead player0)
+          (camera_set_dead (player0))
```
Comment (Mimickal) [inline src/content/h1/engine/scripting/hsc.yml:1752]: Do we need to add parens around `player0`? I have some concerns that the additional parens might actually add to the script syntax nodes in some stupid way. Not sure how to test that.

File: src/content/h1/engine/scripting/hsc.yml [hunk 4]
```diff
@@ -2444,17 +2443,16 @@ Functions:
         en: |-
           ```hsc
           (game_lost)
-
           ```
-          Causes the player to revert to his previous saved game
+          Causes the player to revert to their previous saved checkpoint. For
```
Comment (Mimickal) [inline src/content/h1/engine/scripting/hsc.yml:2447]: :heart:

File: src/content/h1/engine/scripting/hsc.yml [hunk 5]
```diff
@@ -3283,28 +3275,28 @@ Functions:
           (playback)
           ```
           Starts game in film playback mode
-    - slug: player0_joystick_set_is_normal
+    - slug: (player0)_joystick_set_is_normal
```
Comment (Mimickal) [inline src/content/h1/engine/scripting/hsc.yml:3278]: Looks like a find/replace error

File: src/content/h1/guides/map-making/kill-volumes/readme.md [hunk 6]
```diff
@@ -1,14 +1,15 @@
...
 
 # Adding trigger volumes
 The first step is to add trigger volumes to your map's [scenario][] using [Sapien][].
 
 1. Select "Trigger volumes" in the Hierarchy view
 2. Right click on the ground in the Game window to place a volume
 3. Left click and drag the faces of the new volume to adjust its size
-4. Using the Properties palette, give the volume a suitable name like `kill_volume` (should be lower-case)
+4. Rotate the volume using the active marker if needed
+5. Using the Properties palette, give the volume a suitable name like `kill_volume` (should be lower-case)
```
Comment (Mimickal) [inline src/content/h1/guides/map-making/kill-volumes/readme.md:10]: Pretty sure markdown lets you use `1. xyz` for each item in an ordered list, so you can avoid needing to manually re-number list elements like this.

**Review by Mimickal — CHANGES_REQUESTED** [review]

File: src/content/h1/engine/scripting/hsc.yml (hunk 3 above)
Comment (csauve) [reply src/content/h1/engine/scripting/hsc.yml:1752]: We need that -- it's not actually evaluating the static script without them.

File: src/content/h1/guides/map-making/kill-volumes/readme.md (hunk 6 above)
Comment (csauve) [reply src/content/h1/guides/map-making/kill-volumes/readme.md:10]: Whoa, you're right! Nice tip

**Pull request merged**